docs/cas-server-documentation/installation/Rest-Authentication.md

```diff
@@ -45,6 +45,20 @@ Expected responses from the REST endpoint are mapped to CAS as such:
 | `428`          | Produces a `AccountPasswordMustChangeException`
 | Other          | Produces a `FailedLoginException`
 
+## Warnings
+
+The remote REST endpoint can send warnings back to the CAS server using custom headers:
+
+### `X-Cas-PasswordExpirationDate`
+
+If this header is present in the response and contains a RFC1123 date a
+`PasswordExpiringWarningMessageDescriptor` is added to the created `AuthenticationHandlerExecutionResult`.
+
+### `X-Cas-Warning`
+
+For each `X-Cas-Warning` header present in the response, a corresponding `DefaultMessageDescriptor` is added
+to the `AuthenticationHandlerExecutionResult`. The header value is used as `code` and `defaultMessage`.
```

Comment:
Rather that describing what happens internally API-wise (which doesn't really belong here and would serve better as javadocs, etc), it would be much better if you actually describe the *what* and now the *how*. What does CAS do when you pass along the warnings? where are they processed? where do they show up? etc etc. Avoid using class names and try to use plain language as much as possible.